linera-base/src/crypto/secp256k1/evm.rs

```diff
@@ -114,6 +114,34 @@ impl EvmPublicKey {
             }
         }
     }
+
+    /// Returns an EVM address for the public key.
+    pub fn address(&self) -> alloy_primitives::Address {
+        alloy_primitives::Address::from_public_key(&self.0)
+    }
+
+    /// Recover the public key from the signature and the value.
+    ///
+    /// This function turns the `value` into a `CryptoHash`, then hashes it using EIP-191
+    /// and finally recovers the public key from the signature.
+    pub fn recover_from_msg<'de, T>(
+        signature: &EvmSignature,
+        value: &T,
+    ) -> Result<Self, CryptoError>
+    where
+        T: BcsSignable<'de>,
+    {
+        let message = CryptoHash::new(value).as_bytes().0;
+        let public_key =
+            signature
+                .0
+                .recover_from_msg(message)
+                .map_err(|_| CryptoError::InvalidSignature {
+                    error: "Failed to recover public key from signature".to_string(),
+                    type_name: Self::type_name().to_string(),
```

Comment:
(Unrelated, but: Could this be a `&'static str` instead of a `String`?)

Reply:
I am not changing the behavior - this was already here. If we want to change this - I'd suggest doing this in a different PR. This one isn't about it.